spglib 2.4 compatibility

There seems to be a breaking change in spglib 2.4. Euphonic build recipes currently allow spglib 2.4 to be installed, which leads to test failures such as

```
================================================ FAILURES =================================================
_ TestQpointFrequenciesCalculateDosMap.test_calculate_dos_map[quartz-quartz_bandstructure_cv_only_qpoint_frequencies.json-ebins0-quartz_bandstructure_dos_map.json] _

self = <tests_and_analysis.test.euphonic_test.test_qpoint_frequencies.TestQpointFrequenciesCalculateDosMap object at 0x7f8f68b91f90>
material = 'quartz', qpt_freqs_json = 'quartz_bandstructure_cv_only_qpoint_frequencies.json'
ebins = <Quantity([  0.    0.6   1.2   1.8   2.4   3.    3.6   4.2   4.8   5.4   6.    6.6
   7.2   7.8   8.4   9.    9.6  10....5.8 146.4 147.  147.6 148.2 148.8 149.4 150.  150.6
 151.2 151.8 152.4 153.  153.6 154.2 154.8], 'millielectron_volt')>
expected_dos_map_json = 'quartz_bandstructure_dos_map.json'

    @pytest.mark.parametrize(
        'material, qpt_freqs_json, ebins, expected_dos_map_json', [
            ('quartz', 'quartz_bandstructure_cv_only_qpoint_frequencies.json',
             np.arange(0, 155, 0.6)*ureg('meV'),
             'quartz_bandstructure_dos_map.json'),
            ('quartz', 'quartz_bandstructure_cv_only_qpoint_frequencies.json',
             np.concatenate((np.arange(0, 100, 0.3),
                             np.arange(100, 155, 0.6)))*ureg('meV'),
             'quartz_bandstructure_dos_map_uneven_bins.json'),
            ('NaCl', 'NaCl_band_yaml_from_phonopy_qpoint_frequencies.json',
             np.arange(0, 300, 5)*ureg('1/cm'),
             'NaCl_band_yaml_dos_map.json')
        ])
    def test_calculate_dos_map(
            self, material, qpt_freqs_json, ebins, expected_dos_map_json):
        qpt_freqs = get_qpt_freqs(material, qpt_freqs_json)
>       dos_map = qpt_freqs.calculate_dos_map(ebins)

tests_and_analysis/test/euphonic_test/test_qpoint_frequencies.py:573:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
euphonic/qpoint_frequencies.py:278: in calculate_dos_map
    x_data, x_tick_labels = self._get_qpt_axis_and_labels()
euphonic/qpoint_frequencies.py:327: in _get_qpt_axis_and_labels
    x_tick_labels = get_qpoint_labels(
euphonic/util.py:159: in get_qpoint_labels
    xlabels, qpts_with_labels = _recip_space_labels(qpts, cell=cell)
euphonic/util.py:570: in _recip_space_labels
    sym_label_to_coords = seekpath.get_path(cell)["point_coords"]
../../micromamba/envs/euphonic-dev/lib/python3.11/site-packages/seekpath/getpaths.py:182: in get_path
    res = hpkot.get_path(
../../micromamba/envs/euphonic-dev/lib/python3.11/site-packages/seekpath/hpkot/__init__.py:165: in get_path
    dataset = spglib.get_symmetry_dataset(
../../micromamba/envs/euphonic-dev/lib/python3.11/site-packages/spglib/spglib.py:718: in get_symmetry_dataset
    lattice, positions, numbers, _ = _expand_cell(cell)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

cell = (array([[ 2.42617588, -4.20225989,  0.        ],
       [ 2.42617588,  4.20225989,  0.        ],
       [ 0.        ,  0.        ,  5.35030451]]), array([], dtype=float64), array([], dtype=float64))

    def _expand_cell(
        cell: Cell,
    ) -> tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray | None]:
        lattice = np.array(np.transpose(cell[0]), dtype="double", order="C")
        positions = np.array(cell[1], dtype="double", order="C")
        numbers = np.array(cell[2], dtype="intc")
        if len(cell) == 4:
            magmoms = np.array(cell[3], order="C", dtype="double")
        elif len(cell) == 3:
            magmoms = None
        else:
            raise TypeError("cell has to be a tuple of 3 or 4 elements.")

        # Sanity check
        if lattice.shape != (3, 3):
            raise TypeError("lattice has to be a (3, 3) array.")
        if not (positions.ndim == 2 and positions.shape[1] == 3):
>           raise TypeError("positions has to be a (num_atoms, 3) array.")
E           TypeError: positions has to be a (num_atoms, 3) array.
```

Rolling back spglib to version 2.3 fixes this. We could pin the version if necessary, but hopefully it won't be too hard to make things compatible across the supported versions.

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -5,6 +5,12 @@
   - Fixed an error loading QpointPhononModes from JSON when there is a
     single q-point in the data
 
+- Maintenance
+
+  - Compatibility fix for spglib 2.4 update: a new sanity-check in
+    spglib raises TypeError when using empty unit cell and this needs
+    handling when looking for high-symmetry labels
+
 
 -------------------------------------------------------------------------------
 

--- a/euphonic/util.py
+++ b/euphonic/util.py
@@ -568,7 +568,9 @@ def _recip_space_labels(qpts: np.ndarray,
     else:
         try:
             sym_label_to_coords = seekpath.get_path(cell)["point_coords"]
-        except SymmetryDetectionError:
+        except (SymmetryDetectionError, TypeError) as err:
+            if isinstance(err, TypeError):
+                assert "positions has to be" in str(err)
             warnings.warn(('Could not determine cell symmetry, using generic '
                            'q-point labels'), stacklevel=2)
             sym_label_to_coords = _generic_qpt_labels()

--- a/euphonic/util.py
+++ b/euphonic/util.py
@@ -570,7 +570,10 @@ def _recip_space_labels(qpts: np.ndarray,
             sym_label_to_coords = seekpath.get_path(cell)["point_coords"]
         except (SymmetryDetectionError, TypeError) as err:
             if isinstance(err, TypeError):
+                # There is a particular TypeError we expect to see when the
+                # unit cell is empty; make sure we do not have some other error
                 assert "positions has to be" in str(err)
+                assert len(cell[1]) == 0
             warnings.warn(('Could not determine cell symmetry, using generic '
                            'q-point labels'), stacklevel=2)
             sym_label_to_coords = _generic_qpt_labels()

